### ISSUE TITLE
[Bug] Non-optimal regular expression: Suggested fix does not take regex escaping into account

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1066]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug                          |
| Plugin         | Php Inspections (EA Extended), 3.0.6.1 |
| Language level | PHP 7.2                        |

Current behaviour:
---

Sorry, it's me again. 😄 Found one more bug in the "non-optimal regular expression" inspection suggestion.

Here is some example code:

```php
function foo(string $input): string
{
    if (preg_match('/\./', $input)) {
        return 'Input contains a dot.';
    }
    return 'Input does not contain a dot.';
}
```

When using the suggested fix from the "Non-optimal regular expression" inspection ("'false !== strpos($input, "\.")' can be used instead.") then the code is changed into that:

```php
function foo(string $input): string
{
    if (false !== strpos($input, "\.")) {
        return 'Input contains a dot.';
    }
    return 'Input does not contain a dot.';
}
```

Well, obviously this is wrong. The dot was only escaped before, because it has a meaning inside the regex. Now `$input` is not checked for a dot anymore, but insteaad for `\.`.

Expected behaviour:
---

I'd actually expect the suggestion to change the code into this:

```php
function foo(string $input): string
{
    if (strpos($input, '.') !== false) {
        return 'Input contains a dot.';
    }
    return 'Input does not contain a dot.';
}
```

Well I said something about yoda and the double quotes already. But the real bug is that the dot is still being escaped inside `strpos()` from the suggestion.
